### DISCUSSION TITLE
Handling C arrays?

### DISCUSSION THREAD

**Discussion by RazielXYZ** [stephenberry/glaze#879] [Q&A] [answered]
I may be missing something in the documentation, but what (if any) is the recommended way to handle C arrays?
A simple example such as this fails to build. If adding &T::foo manually to the value, it seems to build, but just outputs the pointer, which I suppose is expected, but not exactly the desired outcome here.
```
#include <iostream>
#include <glaze/glaze.hpp>

struct testStruct {
    int foo[20];
};

int main() {
    testStruct t;
    t.foo[0] = 5;

    std::cout << glz::write_json(t);
}
```

**Comment by stephenberry** [comment]
I would probably use a lambda in a `glz::meta` to return a std::span to your data. This would then allow reading/writing to the underlying c style array. I can give a full example when I have access to a computer if you need it.

**Comment by stephenberry** [comment]
Here is an example of using `std::span` and a `glz::meta` to handle your c style array:

```c++
#include <glaze/glaze.hpp>
#include <iostream>
#include <span>

struct testStruct {
    int foo[20];
};

template <>
struct glz::meta<testStruct> {
    using T = testStruct;
    static constexpr auto value =
        object("foo", [](auto& s) { return std::span{ s.foo, 20}; });
};

int main() {
    testStruct t;
    t.foo[0] = 5;
    t.foo[10] = 7;

    std::cout << glz::write_json(t);
}
```

Compiler explorer: https://gcc.godbolt.org/z/v63Y9xKrW

**Comment by stephenberry** [comment, marked as answer]
@RazielXYZ, C style arrays are now directly supported in Glaze for reading/writing

**Reply by RazielXYZ** [reply]
I saw that, think it's a great addition (even though C arrays should probably be avoided when possible)! Thanks for all the continued hard work on both improvements and features!

**Reply by stephenberry** [reply]
You're welcome! I think it makes sense as well, it's especially nice when working on hardware interface structs that use C style arrays.
